### PR TITLE
Define lint codes for more lints

### PR DESCRIPTION
**Pull request by bwilkerson**
Continuing to add explicit lint codes to the lints.

### CODE REVIEW

**Review by pq — APPROVED** [review]
Excellent!

File: lib/src/rules/deprecated_consistency.dart [hunk 1]
```diff
@@ -49,6 +49,20 @@ class B {
 ''';
 
 class DeprecatedConsistency extends LintRule {
+  static const LintCode constructorCode = LintCode('deprecated_consistency',
+      'Constructors in a deprecated class should be deprecated.',
+      correctionMessage: 'Try marking the constructor as deprecated.');
+
+  static const LintCode parameterCode = LintCode('deprecated_consistency',
+      'Parameter that initialize a deprecated field should be deprecated.',
```
Comment (srawlins) [inline lib/src/rules/deprecated_consistency.dart:57]: "initializes"?

Comment (srawlins) [reply lib/src/rules/deprecated_consistency.dart:57]: or "Parameters"

File: lib/src/rules/tighten_type_of_initializing_formals.dart [hunk 2]
```diff
@@ -46,6 +46,11 @@ class C extends B {
 ''';
 
 class TightenTypeOfInitializingFormals extends LintRule {
+  static const LintCode code = LintCode('tighten_type_of_initializing_formals',
+      "Use type annotation rather than 'assert' to enforce non-nullability.",
```
Comment (srawlins) [inline lib/src/rules/tighten_type_of_initializing_formals.dart:50]: "Use a type annotation"?

File: lib/src/rules/use_colored_box.dart [hunk 3]
```diff
@@ -39,13 +39,20 @@ Widget buildArea() {
 ''';
 
 class UseColoredBox extends LintRule {
+  static const LintCode code = LintCode('use_colored_box',
+      "Use 'ColoredBox' rather than a 'Container' with only a 'Color'.",
```
Comment (srawlins) [inline lib/src/rules/use_colored_box.dart:43]: "Use a 'ColoredBox'"?

**Review by srawlins — APPROVED** [review]

**Pull request merged**